components/brave_wallet/browser/solana_keyring.cc

```diff
@@ -57,4 +67,100 @@ std::string SolanaKeyring::GetAddressInternal(HDKeyBase* hd_key_base) const {
   return hd_key->GetBase58EncodedPublicKey();
 }
 
+// Create a valid program derived address without searching for a bump seed.
+// https://docs.rs/solana-program/latest/solana_program/pubkey/struct.Pubkey.html#method.create_program_address
+// static
+absl::optional<std::string> SolanaKeyring::CreateProgramDerivedAddress(
+    const std::vector<std::vector<uint8_t>>& seeds,
+    const std::string& program_id) {
+  const std::string pda_marker = "ProgramDerivedAddress";
+
+  std::vector<uint8_t> program_id_bytes;
+  if (!Base58Decode(program_id, &program_id_bytes, kSolanaPubkeySize))
+    return absl::nullopt;
+
+  if (seeds.size() > kMaxSeeds) {
+    return absl::nullopt;
+  }
+
+  for (const auto& seed : seeds) {
+    if (seed.size() > kMaxSeedLen) {
+      return absl::nullopt;
+    }
+  }
+
+  std::vector<uint8_t> buffer;
+  for (const auto& seed : seeds) {
+    buffer.insert(buffer.end(), seed.begin(), seed.end());
+  }
+
+  buffer.insert(buffer.end(), program_id_bytes.begin(), program_id_bytes.end());
+  buffer.insert(buffer.end(), pda_marker.begin(), pda_marker.end());
+
+  auto hash_array = crypto::SHA256Hash(buffer);
+  std::vector<uint8_t> hash_vec(hash_array.begin(), hash_array.end());
+
+  // Invalid because program derived addresses have to be off-curve.
+  if (bytes_are_curve25519_point(
```

Comment:
I took a look at this aspect of making sure the associate private key is off curve. From what I can tell the purpose of this was so that it can guarantee that the programs can't sign and because this makes it so there's no key management problem needing to be solved for programs.

Essentially, what I gather is they're treating this "public key" which is off curve more like an identifier rather for the Address to keep the pattern of addresses match public keys going without taking up the usable points that do lie on the curve which are usable for proper signatures.